packages/netlify-cms-core/src/actions/editorialWorkflow.ts

```diff
@@ -406,7 +414,10 @@ export function persistUnpublishedEntry(collection: Collection, existingUnpublis
         }),
       );
       dispatch(unpublishedEntryPersisted(collection, transactionID, newSlug));
-      if (!existingUnpublishedEntry) return dispatch(loadUnpublishedEntry(collection, newSlug));
+      if (entry.get('slug') !== newSlug) {
```

Comment:
Originally this was done to navigate from new entry url to the slug url in the editor based on if the entry was new or not. Since the slug can now change for existing entries as well, we use that logic instead.